### PR TITLE
chore(cloud_firestore): upgrade gradle

### PR DESCRIPTION
**Pull request by TarekkMA**
## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/cloud_firestore/cloud_firestore/example/android/app/build.gradle [hunk 1]
```diff
@@ -1,66 +1,61 @@
...
-    kotlinOptions {
-      jvmTarget = '1.8'
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
     }
 
-    sourceSets {
-      main.java.srcDirs += 'src/main/kotlin'
+    defaultConfig {
+        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
```
Comment (Lyokone) [inline packages/cloud_firestore/cloud_firestore/example/android/app/build.gradle:40]: You can remove the TODO

File: packages/cloud_firestore/cloud_firestore/example/android/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.3-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.3-all.zip
```
Comment (russellwheatley) [inline packages/cloud_firestore/cloud_firestore/example/android/gradle/wrapper/gradle-wrapper.properties:5]: Revert this change to gradle version

File: packages/cloud_firestore/cloud_firestore/example/android/app/build.gradle [hunk 3]
```diff
@@ -1,66 +1,61 @@
...
-
-    defaultConfig {
-        applicationId "io.flutter.plugins.firebase.firestore.example"
-        minSdk 21
-        targetSdk 34
-        versionCode flutterVersionCode.toInteger()
-        versionName flutterVersionName
-    }
+    namespace = "io.flutter.plugins.firebase.firestore.example"
+    compileSdk = flutter.compileSdkVersion
```
Comment (russellwheatley) [inline packages/cloud_firestore/cloud_firestore/example/android/app/build.gradle:31]: not sure what flutter `compileSdkVersion` is but we will need this to be a minimum of 34 for android BOM >= 33

**Review by russellwheatley — CHANGES_REQUESTED** [review]
a few things might need change, might be worth waiting until this PR lands before merging this as some things might need changing. But LGTM otherwise.

File: packages/cloud_firestore/cloud_firestore/example/android/app/build.gradle (hunk 3 above)
Comment (TarekkMA) [reply packages/cloud_firestore/cloud_firestore/example/android/app/build.gradle:31]: https://github.com/flutter/flutter/blob/124aacbaef3ff0476f3a02c2ab05271e152224db/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy#L45 

it's should we use it or use hardcoded instead ?

Comment (russellwheatley) [reply packages/cloud_firestore/cloud_firestore/example/android/app/build.gradle:31]: nice, keep it as it is then 👍

**Review by Lyokone — APPROVED** [review]

**Review by russellwheatley — APPROVED** [review]
